Snabble/UI/Payment/PostPaymentManager.swift

```diff
@@ -0,0 +1,158 @@
+        forProject project: Project
+    ) -> Bool
+}
+
+final class PostPaymentManager {
+
+    private var processTimer: Timer?
+    private var sessionTask: URLSessionTask?
+
+    weak var delegate: PostPaymentManagerDelegate?
+
+    let project: Project
+    let checkoutProcess: CheckoutProcess
+
+    init(project: Project, checkoutProcess: CheckoutProcess) {
+        self.project = project
+        self.checkoutProcess = checkoutProcess
+    }
+
+    deinit {
+        cancel()
+    }
+
+    func start() {
+        processTimer?.invalidate()
+        processTimer = Timer.scheduledTimer(
+            withTimeInterval: 1,
+            repeats: false
+        ) { [unowned self] timer in
+            guard timer.isValid else { return }
+            checkoutProcess.update(
+                project,
+                taskCreated: {
+                    sessionTask = $0
+                },
+                completion: {
+                    update(result: $0)
+                })
+        }
+    }
+
+    private func cancel() {
+        processTimer?.invalidate()
+        processTimer = nil
+
+        sessionTask?.cancel()
+        sessionTask = nil
+    }
+
+    // MARK: - process updates
+    private func update(
+        result: RawResult<CheckoutProcess, SnabbleError>
+    ) {
+        switch result.result {
+        case .success(let process):
+            validateProcess(
+                for: process,
+                with: result.rawJson
+            )
+            notifications(
+                for: process,
+                withRawJson: result.rawJson
+            )
+        case .failure(let error):
+            Log.error(String(describing: error))
+            let shouldRetry = delegate?.shouldRetryFailedUpdate(
+                on: self,
+                withCheckoutProcess: checkoutProcess,
+                forProject: project
+            )
+            if shouldRetry ?? true {
+                start()
+            }
+        }
+    }
+
+    private func validateProcess(
+        for process: CheckoutProcess,
+        with rawJson: [String: Any]?
+    ) {
+        let isCompleted: Bool
+        if process.requiresExitToken {
+            isCompleted = process.fulfillmentsDone() && process.exitToken?.value != nil
+        } else {
+            isCompleted = process.fulfillmentsDone()
+        }
+
+        if isCompleted {
+            delegate?.postPaymentManager(
+                self,
+                didCompleteCheckoutProcess: process,
+                withRawJson: rawJson,
+                forProject: project
+            )
+            cancel()
+        } else {
+            delegate?.postPaymentManager(
+                self,
+                didUpdateCheckoutProcess: process,
+                withRawJson: rawJson,
+                forProject: project
+            )
+            start()
+        }
+    }
+
+    private func notifications(for process: CheckoutProcess, withRawJson rawJson: [String: Any]?) {
```

Comment:
würde ich in `postNotifications` o.ä. umbenennen

Reply:
08435f17177e125a7cb4342f024530cb4f10cc4f